src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseGainFlycloak.java

```diff
@@ -14,7 +17,10 @@ public ItemUseGainFlycloak(String[] useParam) {
 
     @Override
     public boolean useItem(UseItemParams params) {
-        params.player.getInventory().addItem(this.i);  // TODO: Currently this returns false for all virtual items - need to have a proper success/fail
+        AvatarFlycloakData flycloakData = GameData.getAvatarFlycloakDataMap().get(this.i);
+        if (flycloakData != null && !params.player.getFlyCloakList().contains(this.i)) {
+            params.player.addFlycloak(this.i);
+        }
```

Comment:
`addFlycloak` is a set operation, it doesn't have ill effects if the set already contains the flycloak.
```suggestion
        if (GameData.getAvatarFlycloakDataMap().containsKey(this.i)) {
            params.player.addFlycloak(this.i);
        }
```